### ISSUE TITLE
feat: scripts/jax_grad/imaging/ — lp.py + mge.py port (task 6/9)

### ISSUE THREAD

**Issue by Jammy2211** [Jammy2211/autogalaxy_workspace_test#28]
## Overview

Port the two autolens `jax_grad/imaging_*.py` scripts to autogalaxy under a new `scripts/jax_grad/imaging/` subfolder. Each script exercises `jax.value_and_grad` on the autogalaxy imaging likelihood path and asserts the gradient is finite, the shape matches the model's free-parameter count, and the gradient isn't all-zero. This is task 6/9 of the autogalaxy_workspace_test parity epic (#5).

## Plan

- Port the two autolens `jax_grad/imaging_*.py` scripts to autogalaxy under a `jax_grad/imaging/` subfolder (autolens is currently flat — surface the layout divergence in the PR body for the maintainer to decide on a follow-up retrofit).
- For each port, keep the autolens script's `jax.value_and_grad` body and assertions verbatim, but swap the model setup for the autogalaxy single-galaxy form already proven by the matching `jax_likelihood_functions/imaging/{lp,mge}.py` scripts on the same `jax_test` dataset.
- Wire the two new entries into `smoke_tests.txt` and add a `jax_grad/` override to `config/build/env_vars.yaml` mirroring the existing `jax_likelihood_functions/` one (full datasets + JAX enabled).

<details>
<summary>Detailed implementation plan</summary>

### Affected Repositories
- autogalaxy_workspace_test (primary)

### Work Classification
Workspace

### Branch Survey

| Repository | Current Branch | Dirty? | Notes |
|---|---|---|---|
| ./autogalaxy_workspace_test | main | dirty (3 modified `dataset/*/jax_test/galaxies.json` + untracked `test_report.md`) | Pre-existing dirt unrelated to this task. Worktree starts clean from `origin/main`. |

**Suggested branch:** `feature/autogalaxy-wst-jax-grad-imaging`
**Worktree root:** `~/Code/PyAutoLabs-wt/autogalaxy-wst-jax-grad-imaging/` (created later by `/start_workspace`)

### Implementation Steps

1. Create `scripts/jax_grad/__init__.py` (empty) and `scripts/jax_grad/imaging/__init__.py` (empty).
2. **`scripts/jax_grad/imaging/lp.py`** — start from `autolens_workspace_test/scripts/jax_grad/imaging_lp.py`. Replace the model construction with the same setup used in `autogalaxy_workspace_test/scripts/jax_likelihood_functions/imaging/lp.py`:
   - `dataset_path = path.join("dataset", "imaging", "jax_test")`, with subprocess fallback to `scripts/jax_likelihood_functions/imaging/simulator.py` if missing.
   - `pixel_scales=0.2`, `mask_radius=3.0`, `over_sample_size_lp=1`.
   - `bulge = af.Model(ag.lp_linear.Sersic)`; `galaxy = af.Model(ag.Galaxy, redshift=0.5, bulge=bulge)`; `model = af.Collection(galaxies=af.Collection(galaxy=galaxy))`.
   - `analysis = ag.AnalysisImaging(dataset=dataset)` (no `positions_likelihood_list` — autogalaxy has no ray-tracing).
   - Keep the autolens reference's `Fitness` setup, the `jnp.array(model.physical_values_from_prior_medians)`, the ell_comps perturbation, the `jax.value_and_grad(fitness.call)` call, and all four assertions (finite value, shape == `model.total_free_parameters`, all-finite grad, not-all-zero grad) verbatim.
3. **`scripts/jax_grad/imaging/mge.py`** — start from autolens `imaging_mge.py`. Mirror `autogalaxy_workspace_test/scripts/jax_likelihood_functions/imaging/mge.py`:
   - Same `jax_test` dataset, `mask_radius=3.0`.
   - `over_sample = ag.util.over_sample.over_sample_size_via_radial_bins_from(grid=dataset.grid, sub_size_list=[4,2,1], radial_list=[0.3,0.6], centre_list=[(0.0,0.0)])`; `dataset.apply_over_sampling(over_sample_size_lp=over_sample)`.
   - `bulge = ag.model_util.mge_model_from(mask_radius=mask_radius, total_gaussians=20, centre_prior_is_uniform=True)`.
   - Single galaxy MGE → `model = af.Collection(galaxies=af.Collection(galaxy=galaxy))`.
   - Same `jax.value_and_grad` body and assertions as `lp.py`.
4. Append two lines to `smoke_tests.txt` immediately after the `jax_likelihood_functions/imaging/` block:
   ```
   jax_grad/imaging/lp.py
   jax_grad/imaging/mge.py
   ```
5. Add a `jax_grad/` override to `config/build/env_vars.yaml` mirroring the existing `jax_likelihood_functions/` one:
   ```yaml
   - pattern: "jax_grad/"
     unset: [PYAUTO_SMALL_DATASETS, PYAUTO_DISABLE_JAX]
   ```
6. Local verify each script: source `activate.sh`, run `python scripts/jax_grad/imaging/lp.py` and `mge.py`. Then exercise via `.github/scripts/run_smoke.py` (`run_one()` for each entry) to confirm the env override applies and the CI shape passes.

### Key Files
- `autogalaxy_workspace_test/scripts/jax_grad/__init__.py` — new (empty namespace marker).
- `autogalaxy_workspace_test/scripts/jax_grad/imaging/__init__.py` — new (empty namespace marker).
- `autogalaxy_workspace_test/scripts/jax_grad/imaging/lp.py` — new port (Sersic linear bulge, single galaxy).
- `autogalaxy_workspace_test/scripts/jax_grad/imaging/mge.py` — new port (MGE linear basis, single galaxy).
- `autogalaxy_workspace_test/smoke_tests.txt` — two new lines.
- `autogalaxy_workspace_test/config/build/env_vars.yaml` — new `jax_grad/` override.

### Risks & Notes
- **autolens layout divergence**: autolens has flat `jax_grad/imaging_lp.py` and `jax_grad/imaging_mge.py`. This PR establishes the subfolder convention (`jax_grad/imaging/{lp,mge}.py`) per the epic spec. The PR body will surface the retrofit question to the maintainer; tasks 7 and 8 (interferometer / multi) compound the divergence.
- **First-run gradient values**: assertions are structural (finite, shape, not-all-zero), not hardcoded literals — no regression-anchor maintenance burden, matching the autolens reference.
- **MGE prior count under SMALL_DATASETS**: same gotcha as task 2 (model_composition); the env override avoids it.
- **Pytree prerequisite**: `_register_fit_imaging_pytrees` is already in place at `autogalaxy/imaging/model/analysis.py:147` (shipped via task 3, exercised by `jax_likelihood_functions/imaging/lp.py`'s JIT round-trip path).

</details>

## Original Prompt

<details>
<summary>Click to expand starting prompt</summary>

Create `scripts/jax_grad/imaging/` in @autogalaxy_workspace_test exercising `jax.grad` on the
autogalaxy imaging likelihood path.

__Important — layout divergence from autolens__

@autolens_workspace_test/scripts/jax_grad/ currently has files **at the top level**
(`imaging_lp.py`, `imaging_mge.py`) — there is no `jax_grad/imaging/` subfolder. The user's
prompt for this epic explicitly asks for `jax_grad/imaging/`, `jax_grad/interferometer/`,
`jax_grad/multi/` subfolders on autogalaxy. Options:

1. Follow the user's directive — use subfolders on autogalaxy even though autolens is flat.
   Flag this drift in the PR description and ask the user whether to retrofit autolens.
2. Ask the user before creating the subfolder.

Recommended: go with (1) — the subfolder layout matches `jax_likelihood_functions/` and is more
extensible. Surface the autolens-retrofit question in the PR body.

__Scripts to port__

From autolens top-level `jax_grad/`:

- `imaging_lp.py` → `jax_grad/imaging/lp.py`
- `imaging_mge.py` → `jax_grad/imaging/mge.py`

Add any further variants (rectangular, delaunay, …) only if the corresponding
`jax_likelihood_functions/imaging/` port from task 3 uncovered a usable `grad`-ready path.
Default: match autolens's coverage one-for-one and leave extras to follow-ups.

**Skip**: anything `*_dspl.py`.

__Pytree prerequisite__

Shares the same `_register_fit_imaging_pytrees` requirement as task 3. By this point the
registration should already be landed on PyAutoGalaxy — if not, block on that first.

__`jax.grad` contract__

Each script prints the gradient vector of a scalar log-likelihood w.r.t. the parameter vector and
asserts it's finite and the shape matches the model's free parameter count. Follow the autolens
reference exactly for the assertions.

__Deliverables__

1. `autogalaxy_workspace_test/scripts/jax_grad/__init__.py`
2. `autogalaxy_workspace_test/scripts/jax_grad/imaging/__init__.py`
3. Ported scripts.
4. Appended to `smoke_tests.txt`.
5. PR body includes the autolens-retrofit question in `## Notes`.

__Depends on__

Task 3 (pytree registration on PyAutoGalaxy imaging analysis).

__Umbrella issue__

Task 6/9. Track under the epic issue on `PyAutoLabs/autogalaxy_workspace_test`.

</details>

**Comment by Jammy2211** [comment]
## Smoke Test Results — 2026-05-05

| Workspace | Passed | Failed | Total |
|-----------|--------|--------|-------|
| autogalaxy_workspace_test (jax_grad/imaging only) | 2 | 0 | 2 |

All 2 smoke tests passed (focused run on new scripts only).

- `jax_grad/imaging/lp.py` — exit 0, gradient shape `(6,)`, all assertions passed
- `jax_grad/imaging/mge.py` — exit 0, gradient shape `(4,)`, all assertions passed

**Comment by Jammy2211** [comment]
## Shipped — 2026-05-05

### PRs
- https://github.com/PyAutoLabs/autogalaxy_workspace_test/pull/29 (merged)

### Summary
Ported the two autolens `jax_grad/imaging_*.py` scripts to autogalaxy under a new `scripts/jax_grad/imaging/` subfolder, with single-galaxy linear Sersic and MGE setups mirroring the matching `jax_likelihood_functions/imaging/*.py` scripts on the same `jax_test` dataset. Both scripts pass in CI on 3.12 (`lp.py` 11.3s, `mge.py` 16.8s).

### Session Notes
- Established the `jax_grad/imaging/` subfolder layout per the parity-epic spec; autolens currently has flat `jax_grad/imaging_lp.py`, `imaging_mge.py`. Open question for the maintainer: should autolens be retrofitted? Filing a follow-up after tasks 7 and 8 land would let the migration cover all three subdirectories at once.
- Pytree registration on `autogalaxy/imaging/model/analysis.py:_register_fit_imaging_pytrees` was already in place from task 3. The `jax.value_and_grad(fitness.call)` path Just Worked end-to-end with no library-side gaps surfaced.
- Added a `jax_grad/` env_vars override mirroring `jax_likelihood_functions/` (unset `PYAUTO_SMALL_DATASETS` + `PYAUTO_DISABLE_JAX`). Without unsetting `PYAUTO_DISABLE_JAX`, the scripts would degrade to the NumPy path and never exercise `jax.grad`; without unsetting `PYAUTO_SMALL_DATASETS`, the MGE basis collapses (same gotcha as task 2).
- Pre-existing CI baseline (already red on main): `jax_likelihood_functions/imaging/rectangular.py`, `interferometer/delaunay_mge.py`, `multi/delaunay_mge.py` on 3.12; broader JAX failures on 3.13. Not introduced by this task.
- Remaining tasks in the parity epic, in numerical order: **7/9 (jax_grad/interferometer)**, **8/9 (jax_grad/multi)**.
